static/js/directives/audiovideo.js

```diff
@@ -117,40 +149,49 @@ define(['jquery', 'underscore', 'text!partials/audiovideo.html', 'text!partials/
 					clonedElement.data("peerid", scope.peerid);
 					scope.element = clonedElement;
 					scope.attachStream = function(stream) {
-						if (stream === dummy) {
+						if (DummyStream.is(stream)) {
+							scope.withvideo = false;
+							scope.onlyaudio = true;
+							_.defer(function() {
+								scope.$emit("active", currentcall);
```

Comment:
Correction, only the `scope.$emit("active", currentcall);` should run in `scope.$apply(...)`.

`$scope.redraw();` sets a flag which is evaluated later by the animation frame and thus doesn't need the angularjs context.